models/userdevice.js

```diff
@@ -11,7 +11,7 @@ var UserDeviceSchema = new Schema({
     iosDeviceToken: {type: String},
     deviceType: {type: String},
     deviceModel: {type: String},
-    deviceId: {type: String},
+    deviceId: {type: String, unique: true},
```

Comment:
So the issue is not that the deviceId is duplicated, but rather entries contain the same 'iosDeviceToken' or 'androidRegistration' field.

Reply:
Uhm... oops, it seems I forgot to commit some changes, where I extended this unique property also to the android registration token as well as the iOS device token. Let me do this now and push it :)